test/test_decomp.py

```diff
@@ -781,7 +787,14 @@ def do_cross_ref(self, device, dtype, op, *, run_all):
                 kwargs = sample_input.kwargs
                 with self.DecompCrossRefMode(self, self.precision, self.rel_tol, dtype, run_all)\
                      as mode, enable_python_dispatcher():
-                    func(*args, **kwargs)
+                    decomposed = func(*args, **kwargs)
+
+                if aten_name in py_impl_decomposition_names:
```

Comment:
This is way too special-cased. We should make this generic to any function that is registered through `py_impl`, rather than manually keeping a list of these functions.

Chances are we should also test properly their backwards.

This change is non-trivial, so it should go in its own PR. You can just prepend a PR to this one.